{"metadata":{"kernelspec":{"name":"ir","display_name":"R","language":"R"},"language_info":{"mimetype":"text/x-r-source","name":"R","pygments_lexer":"r","version":"3.6.0","file_extension":".r","codemirror_mode":"r"},"kaggle":{"accelerator":"none","dataSources":[],"isInternetEnabled":true,"language":"r","sourceType":"notebook","isGpuEnabled":false}},"nbformat_minor":4,"nbformat":4,"cells":[{"cell_type":"code","source":"#BiocManager::install(\"TCGAbiolinks\")\n#BiocManager::install(\"edgeR\")\n#BiocManager::install(\"limma\")\n#BiocManager::install(\"EDASeq\")\nlibrary(\"TCGAbiolinks\")\nlibrary(SummarizedExperiment)\nlibrary(biomaRt)\nlibrary(\"limma\")\nlibrary(\"edgeR\")\nlibrary(\"EDASeq\")\nlibrary(gplots)\n#getProjectSummary(\"TCGA-LUAD\")\n\n#Download and preprocess data\n\nlung_acarcinoma <- GDCquery(project = \"TCGA-LUAD\",\n                data.category = \"Transcriptome Profiling\",\n                data.type = \"Gene Expression Quantification\")\nGDCdownload(lung_acarcinoma)\nlung.data <- GDCprepare(lung_acarcinoma)\nhead(lung.data)\n# Explore Some Metadata Information\nlung.data$definition\nlung.data$cigarettes_per_day\nlung.data$race\ncount(lung.data$ajcc_pathologic_stage==\"Stage IIIA\", na.rm=T)\n#creating metadata table\nslotNames(lung.data)\nall_meta <- data.frame(a=lung.data$paper_expression_subtype)\n\nlung_acarcinoma_meta <- data.frame(\"barcode\"=lung.data$barcode,\n                                   \"sample_type\"=lung.data$definition,\n                         \"cigarettes_per_day\"=lung.data$cigarettes_per_day,\n                         \"race\"=lung.data$race,\n                         \"pathologic_stage\"=lung.data$ajcc_pathologic_stage)\nhead(lung_acarcinoma_meta)\n#filtering metadata for NA containing records\nNA_filtered_data <- na.omit(lung_acarcinoma_meta)\nhead(NA_filtered_data)\n\n#select unstranded dataset\nluad.raw.data <- assays(lung.data) #help to extract information summarizing the experiment\nView(luad.raw.data$unstranded)\n#extracting 20 irregular and 20 regular smoker data from white people having stage I tumor\nselectedBarcodes <- c(subset(NA_filtered_data, race == \"white\" & pathologic_stage %in% c(\"Stage IA\", \"Stage IB\") & sample_type == \"Primary solid Tumor\" & cigarettes_per_day < 1)$barcode[c(1:20)], subset(NA_filtered_data, race == \"white\" & pathologic_stage %in% c(\"Stage IA\", \"Stage IB\") & sample_type == \"Primary solid Tumor\" & cigarettes_per_day > 3)$barcode[c(1:20)])\nselectedData <- luad.raw.data$unstranded[,c(selectedBarcodes)]\nView(selectedData)\n# Data normalization and filtering\nnormData <- TCGAanalyze_Normalization(tabDF = selectedData, geneInfo = geneInfoHT, method = \"geneLength\")\n# filtering the genes with lowest expression\nfiltData <- TCGAanalyze_Filtering(tabDF = normData,\n                                  method = \"quantile\",\n                                  qnt.cut = 0.25)\n# Annotate first 20 columns as 'irregular_smoker'\ncolnames(filtData)[1:20] <- paste0(\"irregular_smoker_\", 1:20)\n\n# Annotate the last 20 columns as 'regular_smoker'\ncolnames(filtData)[21:40] <- paste0(\"regular_smoker_\", 1:20)\n\n#saving the normalized count data table as csv file where first 20 columns are for irregular smoker (<1 cigarette per day) and the last 20 columns are for regular smoker (>2 cigarettes per day) \nwrite.csv(filtData, file = \"Normalized_count_Data_final.csv\", row.names = TRUE)\n#filtData <- read.csv(\"/home/hp/Documents/Stage_3_task/Normalized_count_Data_final.csv\", row.names = 1)\nView(filtData)\n#Differential Expression Analysis\nselectResults <- TCGAanalyze_DEA(\n  mat1 = filtData[, grep(\"^irregular_smoker_\", colnames(filtData))],  # Selecting the 'irregular_smoker' columns\n  mat2 = filtData[, grep(\"^regular_smoker_\", colnames(filtData))],    # Selecting the 'regular_smoker' columns\n  Cond1type = \"Irregular Smoker\",\n  Cond2type = \"Regular Smoker\",\n  pipeline = \"edgeR\",\n  fdr.cut = 0.01,\n  logFC.cut = 2\n)\ndim(selectResults)\nselectResults.level <- TCGAanalyze_LevelTab(selectResults,\"Irregular Smoker\", \"Regular Smoker\",\n                                            filtData[,grep(\"^irregular_smoker_\", colnames(filtData))],\n                                            filtData[,grep(\"^regular_smoker_\", colnames(filtData))])\nView(selectResults.level)\n#heatmap\nheat.data <- filtData[rownames(selectResults.level),]\n#color the plot by the type of smoking status\nsmoking.status <- c(rep(\"Irregular\",20), rep(\"Regular\",20))\ncolor_codes <- c()\nfor (i in smoking.status){\n  if (i == \"Irregular\"){\n    color_codes <- c(color_codes, \"red\")\n  }else{\n    color_codes <- c(color_codes, \"blue\")\n  }\n}\n\n# Now plot the heatmap\nheatmap.2(x=as.matrix(heat.data),\n          col=hcl.colors(10, palette = \"Viridis\"),\n          Rowv = F, Colv = T,\n          scale = \"row\",\n          sepcolor = \"black\",\n          trace = \"none\",\n          key = TRUE,\n          dendrogram = \"col\",\n          cexRow = 0.4, cexCol = 0.8,\n          main = \"Heatmap of DEGs by Smoking Status\",\n          na.color = \"black\",\n          srtCol = 45,\n          ColSideColors = color_codes)\n#Enrichment analysis\n#View the volcano plot first\n# Sample plot: x-axis (logFC), y-axis (-log10(FDR))\nplot(x = selectResults.level$logFC, \n     y = -log10(selectResults.level$FDR),\n     col = ifelse(selectResults.level$logFC > 0, \"blue\", \"red\"),  # Blue for positive, red for negative\n     pch = 16,  # Set point shape (optional)\n     xlab = \"logFC\", \n     ylab = \"-log10(FDR)\",\n     main = \"Volcano Plot with Colored Points\")  # Title of the plot\nupreg.genes <- rownames(subset(selectResults.level, logFC > 2))\ndownreg.genes <- rownames(subset(selectResults.level, logFC < -2))\n\n#convert ensembl IDs to gene IDs using biomart\nmart <- useMart(biomart = \"ensembl\", dataset = \"hsapiens_gene_ensembl\")\nupreg.genes <- getBM(attributes = c(\"ensembl_gene_id\",'hgnc_symbol'),\n                     filters = \"ensembl_gene_id\",\n                     values = upreg.genes,\n                     mart = mart)$hgnc_symbol\ndownreg.genes <- getBM(attributes = c(\"ensembl_gene_id\",'hgnc_symbol'),\n                     filters = \"ensembl_gene_id\",\n                     values = downreg.genes,\n                     mart = mart)$hgnc_symbol\n#enrichment analysis for both\nup.EA <- TCGAanalyze_EAcomplete(TFname=\"Upregulated\",upreg.genes)\ndown.EA <- TCGAanalyze_EAcomplete(TFname=\"Downregulated\",downreg.genes)\nTCGAvisualize_EAbarplot(tf = rownames(up.EA$ResBP),\n                        GOBPTab = up.EA$ResBP,\n                        GOCCTab = up.EA$ResCC,\n                        GOMFTab = up.EA$ResMF,\n                        PathTab = up.EA$ResPat,\n                        nRGTab = upreg.genes,\n                        nBar = 5,\n                        text.size = 2,\n                        fig.width = 30,\n                        fig.height = 15)\nTCGAvisualize_EAbarplot(tf = rownames(down.EA$ResBP),\n                        GOBPTab = down.EA$ResBP,\n                        GOCCTab = down.EA$ResCC,\n                        GOMFTab = down.EA$ResMF,\n                        PathTab = down.EA$ResPat,\n                        nRGTab = downreg.genes,\n                        nBar = 5,\n                        text.size = 2,\n                        fig.width = 30,\n                        fig.height = 15)\n","metadata":{"_uuid":"8f2839f25d086af736a60e9eeb907d3b93b6e0e5","_cell_guid":"b1076dfc-b9ad-4769-8c92-a6c4dae69d19","trusted":true},"execution_count":null,"outputs":[]}]}